lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4414,6 +4463,39 @@ namespace {
             decl, Diagnostic(diag::forward_declared_protocol_label, decl),
             decl->getSourceRange().getBegin());
 
+        if (Impl.ImportForwardDeclarations) {
+          if (auto native = hasNonLocalNativeSwiftDecl<ProtocolDecl>(decl, name, hasKnownSwiftName)) {
```

Comment:
I tried to produce a diagnostic that points at the location of `native`, but the source location on the declaration is invalid. Instead I've added the module that declaration is found in to the new diagnostic note.

If the source location issue is resolved, we can add a better diagnostic.